frontend/src/components/ReferralTable.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useContext, useState } from "react";
+import styled from "styled-components";
+
+import { ReferralTablePagination } from "./ReferralTablePagination";
+import { ReferralTableRow } from "./ReferralTableRow";
+
+import { Referral, getUnitReferrals } from "@/api/units";
+import { AuthContext } from "@/contexts/AuthContext";
+import { DataContext } from "@/contexts/DataContext";
+
+type ReferralTableProps = {
+  id: string;
+};
+
+const ENTRIES_PER_PAGE = 5;
+
+const TableColumnNames = [
+  "Name",
+  "Contact Info",
+  "Referring Staff",
+  "Housing Locator",
+  "Status",
+  "Last Update",
+];
+
+const ReferralTableContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+`;
+
+const ReferralTableTitleSection = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin: 0vh 5vw 0vh 5vw;
+`;
+
+const ReferralTableTitle = styled.h2`
+  color: #000;
+
+  font-family: "Neutra Text";
+  font-size: 32px;
+  font-style: normal;
+  font-weight: 700;
+  line-height: 125%;
+  letter-spacing: 0.64px;
+`;
+
+const ReferralTableButton = styled.button`
+  display: inline-flex;
+  padding: 8px 20px;
+  justify-content: center;
+  align-items: center;
+  gap: 8px;
+
+  border-radius: 12px;
+  border: none;
+  background: var(--Primary, #b64201);
```

Comment:
Just for next time, we don't have variables for these set in our CSS, so you can just use the hex code for the color

i.e.
```css
background: #b64201;
```